### PR TITLE
PinChannelAsyncTests

### PR DESCRIPTION
**Pull request by Gredja**
Tests where created

### CODE REVIEW

File: Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/PinChannelAsyncTests.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+            // Assert
+            act.Should().Throw<NetKitChatNotFoundException>().And.Message.Should()
+                .Be($"Unable to pin channel. Member {nameof(saasUserId)}:{saasUserId} not found.");
+
+            VerifyMocks();
+        }
+
+        [Fact]
+        public void ShouldThrowIfMemberDoesNotInChannel()
```
Comment (eugenekirvel) [inline Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/PinChannelAsyncTests.cs]: ShouldThrowIfMemberIsNotInChannel

File: Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/PinChannelAsyncTests.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+            // Assert
+            act.Should().Throw<NetKitChatInvalidOperationException>().And.Message.Should()
+                .Be($"Unable to pin channel. Member saasUserId:{member.SaasUserId} is not joined channel channelId:{channelId}.");
+
+            VerifyMocks();
+        }
+
+        [Fact]
+        public async Task ShouldReturnTask()
```
Comment (eugenekirvel) [inline Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/PinChannelAsyncTests.cs]: more check

File: Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/PinChannelAsyncTests.cs (hunk 1 above)
Comment (Gredja) [reply Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/PinChannelAsyncTests.cs]: Done

File: Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/PinChannelAsyncTests.cs (hunk 2 above)
Comment (Gredja) [reply Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/PinChannelAsyncTests.cs]: Done

Comment (Gredja) [reply Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/PinChannelAsyncTests.cs]: Add await in Act region

File: Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/PinChannelAsyncTests.cs [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            _memberRepositoryMock.Setup(x => x.GetMemberBySaasUserIdAsync(It.IsAny<string>()))
+                .ReturnsAsync((Member)null)
+                .Verifiable();
+
+            // Act
+            Func<Task> act = async () => await _channelService.PinChannelAsync(saasUserId, channelId, true);
+
+            // Assert
+            act.Should().Throw<NetKitChatNotFoundException>().And.Message.Should()
+                .Be($"Unable to pin channel. Member {nameof(saasUserId)}:{saasUserId} not found.");
```
Comment (eugenekirvel) [inline Softeq.NetKit.Chat.Tests/Unit/Domain.Services/ChannelService/PinChannelAsyncTests.cs]: is not found

**Pull request merged**